ctapipe/io/simteleventsource.py

```diff
@@ -382,7 +380,7 @@ def _generate_events(self):
                 mc.pedestal = array_event["camera_monitorings"][tel_id]["pedestal"]
                 mc.true_image = (
                     array_event.get("photoelectrons", {})
-                    .get(tel_index, {})
+                    .get(tel_id - 1, {})
```

Comment:
I think this should be just tel_id, why did you add the `-1`?

Reply:
for my case array_event['photoelectrons'] has keys which are from [2, 3, 4, 5] whereas file_.header['tel_id'].tolist() = [3,4,5,6].
From that observation the -1 is needed and I guess the difference comes from different indexes conventions.

Reply:
Yes, I can reproduce this. But this shouldn't be the case. I asked Konrad about this. Only the 1208 objects with the photo electrons seem to have this id off by one error.

Reply:
Relevant previous PR: #943

Reply:
I reiterate the need to have a test file with the photoelectron image inside, to test this behaviour and ensure it isn't broken

Reply:
We have that now. It's the `lst_prod3_...` file. But that does not have ignored telescopes. So tel_index == tel_id - 1.


Here we have another case. What is used here as `tel_id` in the photo electron block seems to be the **initial** telescope index before telescopes where ignored.

It would be interesting what's in there when telescopes are simulated with non-consecutive ids and then some are ignored.